### ISSUE TITLE
Support for Rule and Example reserved words

### ISSUE THREAD

**Issue by cquirosj** [SpecFlowOSS/SpecFlow.Rider#21]
Gherkin 6 (supported by Specflow) adds support for the optional Keyword **Rule:** even more importantly it also adds support for the word **Example:** as a synonym of Scenario.

https://docs.specflow.org/projects/specflow/en/latest/Gherkin/Gherkin-Reference.html#keywords

While using a feature file where scenarios are organized using the Scenario reserved word the Rider Plug-in works fine, but it doesn't work when "Scenarios" are organized in the form of Rule: and Example: 

Here is the content of a sample feature file that can be used to reproduce the issue:

Feature: Domestic transfers

    Background:
        Given Peter has an active current account
        And Tracy has an account in another bank with the following details:
        | Name                   | Address Line 1 | City              | State | Zip   | Account Number | Routing Number |
        | Tracy Transferee | 1 Main Street  | Transferville | CA     | 9100| 123456788898   | 123456235         |

    Rule: Domestic transfer in USD require 2-factor authentication
        Example: Peter transfers $5000 to Tracy
            When Peter requests a wire transfer of $5000 to Tracy
            And he agrees to the terms and conditions
            And he completes the 2-factor authentication
            Then Peter should be informed that the transfer has been submitted
            And he should receive a confirmation email containing his transfer details

    Rule: Customers must agree to terms and conditions and complete TFA to make a transfer
        Example: Peter doesn't approve the terms and conditions
            When Peter requests a wire transfer of $5000 to Tracy
            But he doesn't not agree to the terms and conditions
            Then Peter should not be allowed to make the transfer

        Example: Peter doesn't complete the TFA
            When Peter requests a wire transfer of $5000 to Tracy
            And he agrees to the terms and conditions
            But he doesn't complete the 2-factor authentication
            Then Peter should not be allowed to make the transfer

**Comment by Socolin** [comment]
This should be fixed in 1.3.5.

(I also found other bugs in step generation thanks to your example)

**Comment by cquirosj** [comment]
Thank you! that was fast. I installed and manually tested it, I see the plug-in detects missing step definitions and offers me the action to create them. 

I noticed another small thing and thought about reporting them, the **RunTestInGutter**: https://raw.githubusercontent.com/SpecFlowOSS/SpecFlow.Rider/master/doc/images/RunTestInGutter.png ... is only working for the Feature: keyword but it is not working for the grouping of Rule: nor for the Example: reserved word. 

It works fine if the Scenario uses the "**Scenario**" reserved word instead of the synonym "**Example**" and as long as the Scenario is not "nested"/preceded by a Rule section.

**Comment by Socolin** [comment]
> I noticed another small thing and thought about reporting them, the **RunTestInGutter**: https://raw.githubusercontent.com/SpecFlowOSS/SpecFlow.Rider/master/doc/images/RunTestInGutter.png ... is only working for the Feature: keyword but it is not working for the grouping of Rule: nor for the Example: reserved word.

This should be fixed ( I already commit a fix this morning 1h ago) but since you just tests it, can you check if you used the latest version https://github.com/SpecFlowOSS/SpecFlow.Rider/commit/e8e2271316aec94694cefbf834e54113ace91369 or was it an earlier one ?

**Comment by cquirosj** [comment]
Thank you, I confirm it works now. Earlier, I used 1.3.5 binaries delivered by Rider UI, this time I downloaded and manually installed the latest binaries produced by CI and it is working as expected. 

I tried to reproduce the error by installing 1.3.5 this third time again using JetBrains Market place, https://plugins.jetbrains.com/plugin/15957-specflow-support but it worked fine, I was unable to reproduce it.

After every install, I restarted the IDE. Anyways, I glad the run from gutter works too. Thanks again!
